src/test/java/io/github/wistefan/mapping/desc/EntityVOMapperTest.java

```diff
@@ -375,6 +375,25 @@ void testNotificationEndpointMapping() {
                 "The pojo should have the same notification endpoint");
     }
 
+    @DisplayName("Map entity with duplicate relationship")
+    @Test
+    void testDuplicateRelationship() throws Exception {
+        MySubPropertyEntity expectedSubEntity = new MySubPropertyEntity("urn:ngsi-ld:sub-entity:the-sub-entity");
+        MyPojoWithSubEntityList expectedPojo = new MyPojoWithSubEntityList("urn:ngsi-ld:complex-pojo:the-test-pojo");
+        expectedPojo.setMySubPropertyList(List.of(expectedSubEntity, expectedSubEntity));
```

Comment:
I am confused: what I understood is that the duplicates are filtered out, so shouldn't the expectedPojo list should only have one entry?

Reply:
No. They are just filtered out in the interemediate list of resolved realtionships that are used to fill the mapped object. To reduce the requests to the broker, all of them are resolved before actually constructing the object and held in the map. In this map, every relationship can only exist once. However, the object itself is allowed to provide multiple instances of a relationship.